### DISCUSSION TITLE
Tenant specific logging

### DISCUSSION THREAD

**Discussion by christopherarter** [spatie/laravel-multitenancy#406] [Ideas]
This may be a feature I'm not aware of, but I didn't see it when I searched the docs. We have a use case for knowing which log messages were generated by which tenant. We've created a helper, `tenant_log_info(...)`, `tenant_log_error(...)` etc. for each log level. This prefixes the log with the given current tenant. So if the current tenant is `Foo`, by calling `tenant_log_info('Logging some information here')` we would get:

```
[2022-10-21 18:16:23] local.INFO: [Tenant: Foo] Logging some information here
```
Would this be of use to anyone in the base package itself? A PR for this would likely just be a `helpers` file with some helpers like above. I'd be happy to submit one if others see a use case. It would be a pretty simple wrapper around the existing helpers with a tenant prefix'd.

Just an idea :)

**Comment by chris-ware** [comment]
For my purpose, I've just set the shared the tenant into the logging context like so:

```php
class AddLoggingContext implements SwitchTenantTask
{
    public function makeCurrent(Tenant $tenant): void
    {
        Log::shareContext(array_filter([
            'tenant' => $tenant->name,
            'url' => url()->current(),
        ]));
    }

    public function forgetCurrent(): void
    {
        Log::withoutContext();
    }
}
```
